testing/src/cypress/e2e/02-institution/07-location-programs.cy.ts

```diff
@@ -1,47 +1,155 @@
-import DashboardInstitutionObject from "../../page-objects/Institution-objects/DashboardInstitutionObject";
-import ManageUsersObject from "../../page-objects/Institution-objects/ManageUsersObject";
+import DashboardInstitutionObject, {
+  CredentialTypes,
+} from "../../page-objects/Institution-objects/DashboardInstitutionObject";
 import LocationProgramObject from "../../page-objects/Institution-objects/LocationProgramObject";
-import InstitutionCustomCommand from "../../custom-command/institution/InstitutionCustomCommand";
-import { programData } from "../../../data/dev/institution-data/institutionProgramData";
 import InstitutionHelperActions from "../../custom-command/institution/common-helper-functions.cy";
+import ProgramDetailViewObject from "../../page-objects/Institution-objects/ProgramDetailViewObject";
+import data from "../../e2e/data/institution/manage-location.json";
 
 const dashboardInstitutionObject = new DashboardInstitutionObject();
-const manageUsersObject = new ManageUsersObject();
 const locationProgramObject = new LocationProgramObject();
 const institutionHelperActions = new InstitutionHelperActions();
+const programDetailViewObject = new ProgramDetailViewObject();
 
 describe("Location Program", () => {
-  beforeEach(() => {
+  before(() => {
     institutionHelperActions.loginIntoInstitutionSingleLocation();
   });
 
+  beforeEach(() => {
+    dashboardInstitutionObject.homeButton().click();
+    dashboardInstitutionObject.clickOnSideBar("Vancouver", "Programs");
+  });
+
   it("Verify that user redirect to location program page", () => {
-    dashboardInstitutionObject.dashboardButton().click();
-    dashboardInstitutionObject.programsButton().eq(0).click();
+    dashboardInstitutionObject.institutionProgramsHeader().should("be.visible");
     cy.url().should("contain", "/location-programs");
   });
 
-  it("Verify that search bar is working properly by searching incorrect word", () => {
-    dashboardInstitutionObject.dashboardButton().click();
-    dashboardInstitutionObject.programsButton().eq(0).click();
-    manageUsersObject.searchUserInputText().type("Dummy data");
-    manageUsersObject.searchButton().click();
+  it("Verify that user when searched for incorrect programs will return correct results", () => {
+    dashboardInstitutionObject
+      .institutionSearchInputBox()
+      .type("Dummy")
+      .type("{enter}");
+    dashboardInstitutionObject.noProgramsYetText().should("be.visible");
+  });
+
+  it("Verify that user when searched for programs will display correct results", () => {
+    dashboardInstitutionObject
+      .institutionSearchInputBox()
+      .type("Testing")
```

Comment:
A valid program name can come from a constant.

Reply:
Sure - we can update them as needed when we end up creating the programs using data seed